### PR TITLE
Add Teleport loadtest infrastructure and grafana dashboard

### LINKED ISSUE #8709: Open Source Grafana Dashboard
Update and open source Grafana dashboard used internally for load testing.

### PR DESCRIPTION
**Pull request by rosstimothy**
This adds loadtesting automation and k8s yamls to deploy teleport. The grafana
health dashboard is also included and can be consumed by end users to better monitor
their teleport clusters.

Closes #8709

### CODE REVIEW

**Comment by russjones** [comment]
@tcsc You might be interested in this as well.

File: assets/loadtest/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	@make -C k8s run-soak-tests SOAK_TEST_DURATION=$(SOAK_TEST_DURATION)	
+
+# run 500 node trusted cluster scaling test
+.PHONY: run-tc-scaling-test
+run-tc-scaling-test:
+	@make -C k8s run-tc-scaling-test
+
+.PHONY: run-scaling-test
+run-scaling-test:
+	@make -C k8s run-scaling-test
```
Comment (r0mant) [inline assets/loadtest/Makefile:46]: Nit: Might be useful to expand descriptions for these targets a bit to explain what kind of tests they actually run.

File: assets/loadtest/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+You must supply an [OIDC Connector](https://goteleport.com/docs/enterprise/sso/oidc/) that will be used for authentication. Create [`oidc.yaml`](./teleport/oidc.yaml)
+before attempting to deploy Teleport to the cluster.
+
+
+## Deploy Teleport
+
+### etcd Backend
+
+```bash
+$ make delpoy-etcd-cluster
```
Comment (r0mant) [inline assets/loadtest/README.md:74]:
```suggestion
$ make deploy-etcd-cluster
```

File: assets/loadtest/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+### Firestore Backend
+
+To use the firestore backend you must have a GCP service account key with `Cloud Datastore User`, `Cloud Datastore Index Admin`
+`Storage Object Creator`, `Storage Object Viewer`. Set `GCP_CREDS_LOCATION` to the location that you saved the service account key.
+
+```bash
+$ export GCP_CREDS_LOCATION=/path/to/service/account/key
+$ make delpoy-firestore-cluster
```
Comment (r0mant) [inline assets/loadtest/README.md:84]:
```suggestion
$ make deploy-firestore-cluster
```

File: assets/loadtest/cluster/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# cluster
+
+Automation for creating the `loadtest` kuberentes cluster.
```
Comment (r0mant) [inline assets/loadtest/cluster/README.md:3]:
```suggestion
Automation for creating the `loadtest` kubernetes cluster.
```

File: assets/loadtest/etcd/certs/Makefile [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# output:
+#   ca-key.pem      : private key of the trusted CA
+#   ca-cert.pem     : self-signed CA cert
+#   client-crt.pem  : client cert signed by CA
+# 	client-key.pem  : private key of the client
+.PHONY:all
+all: ca-cert.pem server-cert.pem client-cert.pem
+	@rm -rf *csr
```
Comment (r0mant) [inline assets/loadtest/etcd/certs/Makefile:10]: Nit: I'm curious why removing CSR's is not a part of those respective targets-dependencies (seems like they're already removing intermediate files).

File: assets/loadtest/etcd/certs/Makefile [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		-out client.csr
+	# sign it with CA:
+	@touch index.txt
+	@echo '03' > serial
+	openssl ca -batch \
+		-extensions etcd_client \
+		-config openssl.cnf \
+	    -keyfile ca-key.pem \
+	    -cert ca-cert.pem \
+	    -out client-cert.pem \
```
Comment (r0mant) [inline assets/loadtest/etcd/certs/Makefile:28]: Indentation here seems off.

File: assets/loadtest/etcd/certs/Makefile [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+		-out server.csr
+	# sign it with CA:
+	@touch index.txt
+	@echo '01' > serial
+	openssl ca -batch \
+		-extensions etcd_server \
+		-config openssl.cnf \
+	    -keyfile ca-key.pem \
+	    -cert ca-cert.pem \
+	    -out server-cert.pem \
```
Comment (r0mant) [inline assets/loadtest/etcd/certs/Makefile:48]: Indentation is off.

File: assets/loadtest/k8s/Makefile [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+USE_CERT_MANAGER ?= yes
+
+# performs initialization needed for cluster
+# 1) generates etcd certs
+# 2) generates credentials for grafana and influx
+# 2) creates loadtest namespace
+# 3) installs cert-manager
+# 4) creates and applies secrets
+.PHONY: setup
+.PHONY: setup
```
Comment (r0mant) [inline assets/loadtest/k8s/Makefile:14]:
```suggestion
.PHONY: setup
```

**Review by r0mant — APPROVED** [review]

File: assets/loadtest/etcd/certs/Makefile (hunk 5 above)
Comment (rosstimothy) [reply assets/loadtest/etcd/certs/Makefile]: This came from the original loadtest [branch](https://github.com/gravitational/ops/blame/fspmarshall/monster/loadtest/etcd/certs/Makefile#L10). Moved to the respective targets.

**Review by russjones — APPROVED** [review]

**Comment by russjones** [comment]
@rosstimothy Approved, let's get this merged in.

**Pull request merged**